### PR TITLE
WIP: Added more documentation and maked some functions as private

### PR DESCRIPTION
**Pull request by sigmen**

### CODE REVIEW

File: lib/exceptions.ex [hunk 1]
```diff
@@ -1,4 +1,8 @@
 defmodule OperationError do
+  @moduledoc """
+  Defines an exception that handles an inaccessible wsdl operations.
```
Comment (Nitrino) [inline lib/exceptions.ex]:
```suggestion
  Defines an exception that handles not described WSDL operations.
```

File: lib/soap.ex [hunk 2]
```diff
@@ -1,12 +1,33 @@
 defmodule Soap do
   @moduledoc """
-  Provides a functions for send SOAP requests.
+  The SOAP client for Elixir based on HTTPoison (for send requests) and SweetXml (for xml parsing).
```
Comment (Nitrino) [inline lib/soap.ex]: I think there would be better to talk about the modules of the library and show the flow of use.
e.g. https://github.com/elixir-ecto/ecto/blob/master/lib/ecto.ex

File: lib/soap.ex [hunk 3]
```diff
@@ -24,14 +45,21 @@ defmodule Soap do
   def init_model(path, :url), do: Wsdl.parse_from_url(path)
 
   @doc """
-  Sends a request to the SOAP server based on the passed wsdl_model, action and parameters.
+  Send a request to the SOAP server based on the passed wsdl, action and parameters.
```
Comment (Nitrino) [inline lib/soap.ex]:
```suggestion
  Send a request to the SOAP server based on the passed WSDL file, action and parameters.
```

File: lib/soap.ex [hunk 4]
```diff
@@ -24,14 +45,21 @@ defmodule Soap do
...
+  Returns `{:ok, %Soap.Response{}}` if the request is successful, {:error, reason} otherwise.
 
   ## Parameters
 
   - `wsdl`: Wsdl model from `Soap.init_model/2` function.
   - `action`: Soap action to be called.
```
Comment (Nitrino) [inline lib/soap.ex]:
```suggestion
  - `action`: Soap action to be called. Use `Soap.operation/1` to get a list of available actions 
```

File: lib/soap.ex [hunk 5]
```diff
@@ -24,14 +45,21 @@ defmodule Soap do
...
   - `params`: Parameters for build the body of a XML request.
   - `headers`: Custom request headers.
+  - `opts`: HTTPoison options
```
Comment (Nitrino) [inline lib/soap.ex]:
```suggestion
  - `opts`: HTTPoison options.
```

File: lib/soap.ex [hunk 6]
```diff
@@ -24,14 +45,21 @@ defmodule Soap do
...
   @doc """
-  Sends a request to the SOAP server based on the passed wsdl_model, action and parameters.
+  Send a request to the SOAP server based on the passed wsdl, action and parameters.
+  Returns `{:ok, %Soap.Response{}}` if the request is successful, {:error, reason} otherwise.
 
   ## Parameters
 
   - `wsdl`: Wsdl model from `Soap.init_model/2` function.
   - `action`: Soap action to be called.
   - `params`: Parameters for build the body of a XML request.
```
Comment (Nitrino) [inline lib/soap.ex]:
```suggestion
  - `params`: Parameters to build the body of a SOAP request.
```

**Comment by sigmen** [comment]
fixed in https://github.com/potok-digital/soap/pull/60

**Pull request merged**